backend/src/test/java/com/qwarty/auth/service/GuestServiceTest.java

```diff
@@ -0,0 +1,34 @@
+package com.qwarty.auth.service;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import com.qwarty.auth.dto.GuestAuthResponseDTO;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.test.context.bean.override.mockito.MockitoSpyBean;
+
+@SpringBootTest
+public class GuestServiceTest {
+
+    @Autowired
+    private GuestService guestService;
+
+    @MockitoSpyBean
+    private JwtService jwtService;
+
+    @Test
+    void continueAsGuest_successful() {
+        GuestAuthResponseDTO response = guestService.continueAsGuest();
+
+        assertTrue(
+                response.username().matches("[A-Za-z]+[A-Za-z]+#\\d{3}"),
```

Comment:
```suggestion
                response.username().matches("([A-Z][a-z]+){2}#\d{3}"),
```

This should satisfy the expected pattern more strictly

<img width="1224" height="679" alt="Image" src="https://github.com/user-attachments/assets/30de934c-bcfe-4689-aa72-d62d549efe42" />

Reply:
ooh nice catch, ty ty!